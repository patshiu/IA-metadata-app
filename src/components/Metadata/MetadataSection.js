import React from 'react';
import PropTypes from 'prop-types';
import SectionTitle from '../SectionTitle';
import MetadataList from './MetadataList';

import './MetadataSection.scss';

const MetadataSection = ({ metadata }) => {
  return (
    <div className="row">
      <div className="item-metadata-full-list section padded">
        <SectionTitle title="Full Metadata" count={Object.keys(metadata).length} />
        <MetadataList metadata={metadata} />
      </div>
    </div>
  );
};

MetadataSection.propTypes = {
  metadata: PropTypes.object
};

export default MetadataSection;
