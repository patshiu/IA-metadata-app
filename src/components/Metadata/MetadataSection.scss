@import '../../_variables';

.item-metadata-full-list {
  border: 1px solid $c-light;
  background-color: $base-darker;
  padding: 1rem 2rem 4rem 2rem;
  border-radius: $border-radius-sm;
}
