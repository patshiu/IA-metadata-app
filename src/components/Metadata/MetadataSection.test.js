import React from 'react';
import renderer from 'react-test-renderer';

import MetadataSection from './MetadataSection';

it('renders item metadata correctly', () => {
  const metadata = {
    mediatype: 'movies',
    identifier: 'InformationM',
    publisher: 'Eames (Charles And Ray)',
    description:
      "Applies graphic sensitivity to medium in cartoon form, and traces the history of storing and analyzing information from the days of the cavemen to today's age of electronic brains.",
    date: '1958',
    licenseurl: 'http://creativecommons.org/licenses/publicdomain/',
    color: 'color',
    sound: 'sound',
    collection: 'prelinger',
    title: 'Information Machine, The',
    sponsor: 'IBM',
    pick: '0',
    runtime: '00:09:44',
    updatedate: '2005-01-13 09:36:44',
    updater: 'AV Geek Skip',
    subject: 'need keyword',
    numeric_id: '3254',
    type: 'MovingImage',
    proddate: '1958',
    collectionid: 'InformationM',
    publicdate: '2007-03-01 10:46:58',
    backup_location: 'ia903600_31',
    closed_captioning: 'no',
    addeddate: '2007-03-01 10:46:58'
  };
  const tree = renderer.create(<MetadataSection metadata={metadata} />).toJSON();
  expect(tree).toMatchSnapshot();
});
