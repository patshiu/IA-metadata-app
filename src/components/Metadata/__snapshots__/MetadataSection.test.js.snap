// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders item metadata correctly 1`] = `
<div
  className="row"
>
  <div
    className="item-metadata-full-list section padded"
  >
    <h2>
      Full Metadata
       
      (24)
    </h2>
    <dl>
      <div>
        <dt>
          mediatype
        </dt>
        <dd>
          movies
        </dd>
      </div>
      <div>
        <dt>
          identifier
        </dt>
        <dd>
          InformationM
        </dd>
      </div>
      <div>
        <dt>
          publisher
        </dt>
        <dd>
          Eames (Charles And Ray)
        </dd>
      </div>
      <div>
        <dt>
          description
        </dt>
        <dd>
          Applies graphic sensitivity to medium in cartoon form, and traces the history of storing and analyzing information from the days of the cavemen to today's age of electronic brains.
        </dd>
      </div>
      <div>
        <dt>
          date
        </dt>
        <dd>
          1958
        </dd>
      </div>
      <div>
        <dt>
          licenseurl
        </dt>
        <dd>
          http://creativecommons.org/licenses/publicdomain/
        </dd>
      </div>
      <div>
        <dt>
          color
        </dt>
        <dd>
          color
        </dd>
      </div>
      <div>
        <dt>
          sound
        </dt>
        <dd>
          sound
        </dd>
      </div>
      <div>
        <dt>
          collection
        </dt>
        <dd>
          prelinger
        </dd>
      </div>
      <div>
        <dt>
          title
        </dt>
        <dd>
          Information Machine, The
        </dd>
      </div>
      <div>
        <dt>
          sponsor
        </dt>
        <dd>
          IBM
        </dd>
      </div>
      <div>
        <dt>
          pick
        </dt>
        <dd>
          0
        </dd>
      </div>
      <div>
        <dt>
          runtime
        </dt>
        <dd>
          00:09:44
        </dd>
      </div>
      <div>
        <dt>
          updatedate
        </dt>
        <dd>
          2005-01-13 09:36:44
        </dd>
      </div>
      <div>
        <dt>
          updater
        </dt>
        <dd>
          AV Geek Skip
        </dd>
      </div>
      <div>
        <dt>
          subject
        </dt>
        <dd>
          need keyword
        </dd>
      </div>
      <div>
        <dt>
          numeric_id
        </dt>
        <dd>
          3254
        </dd>
      </div>
      <div>
        <dt>
          type
        </dt>
        <dd>
          MovingImage
        </dd>
      </div>
      <div>
        <dt>
          proddate
        </dt>
        <dd>
          1958
        </dd>
      </div>
      <div>
        <dt>
          collectionid
        </dt>
        <dd>
          InformationM
        </dd>
      </div>
      <div>
        <dt>
          publicdate
        </dt>
        <dd>
          2007-03-01 10:46:58
        </dd>
      </div>
      <div>
        <dt>
          backup_location
        </dt>
        <dd>
          ia903600_31
        </dd>
      </div>
      <div>
        <dt>
          closed_captioning
        </dt>
        <dd>
          no
        </dd>
      </div>
      <div>
        <dt>
          addeddate
        </dt>
        <dd>
          2007-03-01 10:46:58
        </dd>
      </div>
    </dl>
  </div>
</div>
`;
